### PR TITLE
Reactify Homepage

### PR DESCRIPTION
**Pull request by gvn**
Related to #656

### CODE REVIEW

**Comment by gvn** [comment]
There's a few tickets for backend that need to land before this is merged and that affect testing a bit:

https://github.com/mozilla/network/issues/769
https://github.com/mozilla/network/issues/761

Basically: only 3 people show up right now (4 should) and one glyph image will be broken.

File: source/js/components/highlights/highlights.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export default class Highlights extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+  }
+
+  render() {
+    // Limit to 2 additional highlights
+    let highlights = this.props.data.slice(1, 3).map((item, index, array) => {
```
Comment (mmmavis) [inline source/js/components/highlights/highlights.jsx]: we should type check `this.props.data` and probably also give it a default value `[]`

File: source/js/components/highlights/highlights.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      return (
+        <div key={index}>
+          <h5 className="h4-light-black">{item.title}</h5>
+          <p className="body-black">{item.description}</p>
+          <a className="cta-link" href={item.link_url}>{item.link_label}</a>
+          { index < array.length && <hr/> }
+        </div>
+      );
+    });
+
```
Comment (mmmavis) [inline source/js/components/highlights/highlights.jsx]: I would probably create a var for `this.props.data[0]`

File: source/js/components/home-news/home-news.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          </h5>
+          { item.author && <p className="italic-black mb-1">by {item.author}</p> }
+          <p className="small-gray">{moment(item.date, `YYYY-MM-DD`).format(`MMMM YYYY`)}</p>
+          { hr && <hr/> }
+        </div>
+      );
+
+    };
+
+    let unfeaturedNews = this.props.data.slice(1).map((item, index, array) => {
```
Comment (mmmavis) [inline source/js/components/home-news/home-news.jsx]: again, let's type check `this.props.data` and set a default value for it

File: source/js/components/home-news/home-news.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            <a href={item.link} className={featured ? `h4-medium-black` : `h5-black`}>{item.headline}</a>
+          </h5>
+          { item.author && <p className="italic-black mb-1">by {item.author}</p> }
+          <p className="small-gray">{moment(item.date, `YYYY-MM-DD`).format(`MMMM YYYY`)}</p>
+          { hr && <hr/> }
+        </div>
+      );
+
+    };
+
```
Comment (mmmavis) [inline source/js/components/home-news/home-news.jsx]: Should we `let featuredNews = this.props.data[0]`?

File: source/js/components/leaders/leaders.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+export default class Leaders extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {};
+  }
+
+  render() {
+    // Limit to 4 leaders
+    let leaders = this.props.data.slice(0, 4).map((item, index) => {
```
Comment (mmmavis) [inline source/js/components/leaders/leaders.jsx]: let's type check `this.props.data` and set a default value for it

File: source/js/main.js [hunk 6]
```diff
@@ -11,13 +11,35 @@ import People from './components/people/people.jsx';
...
+  fetchData (callback) {
+    // Only fetch data if you're on the homepage
+    if (document.querySelector(`#view-home`)) {
+      let homepageReq = new XMLHttpRequest();
+
+      homepageReq.addEventListener(`load`, () => {
+        callback.call(this, JSON.parse(homepageReq.response));
+      });
+
+      // TODO: Change to env based URL
```
Comment (mmmavis) [inline source/js/main.js]: we can remove this comment now?

File: source/js/main.js [hunk 7]
```diff
@@ -85,7 +107,7 @@ let main = {
   },
 
   // Embed various React components based on the existence of containers within the current page
-  injectReactComponents () {
+  injectReactComponents (data) {
```
Comment (mmmavis) [inline source/js/main.js]: `date = {}`?

**Pull request merged**